### PR TITLE
Fix simple typescript errors

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?

This PR fixes the typescript errors in:

- `assets/pages/contributions-landing/components/AmazonPay/AmazonPayWallet.tsx":1"assets/pages/contributions-landing/components/StripeCardForm/StripeCardForm.tsx`
- `assets/pages/digital-subscription-checkout/thankYouContainer.ts`        
- `assets/pages/digital-subscription-checkout/thankYouGift.tsx`
- `assets/pages/paper-subscription-landing/paperSubscriptionLandingPage.tsx`
- `assets/pages/weekly-subscription-landing/components/weeklyProductPrices.tsx`
- `assets/components/datePicker/datePickerHelpers.test.ts`
- `assets/helpers/rendering/ssrPages.ts`
- `assets/pages/contributions-landing/components/ExistingRecurringContributorErrorMessage.tsx`
- assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.test.tsx`

### CODE REVIEW

File: support-frontend/assets/components/datePicker/datePickerHelpers.test.ts [hunk 1]
```diff
@@ -10,13 +10,13 @@ describe('datePickerHelpers', () => {
...
-		constructor(param) {
-			super(param || '2020-05-14T11:01:58.135Z');
+		constructor(param?: number | string | Date) {
+			super(param ?? '2020-05-14T11:01:58.135Z');
 		}
 	}
 
 	beforeEach(() => {
-		global.Date = MockDate;
+		global.Date = MockDate as DateConstructor;
```
Comment (TomPretty) [inline support-frontend/assets/components/datePicker/datePickerHelpers.test.ts:19]: Just a test file so I think it's fine to use type assertions

File: support-frontend/assets/helpers/rendering/ssrPages.ts [hunk 2]
```diff
@@ -2,15 +2,13 @@ import { renderToString } from 'react-dom/server';
 import { EmptyContributionsLandingPage } from 'pages/contributions-landing/EmptyContributionsLandingPage';
 import { content as showcase } from 'pages/showcase/showcase';
 
-const render = (content) => renderToString(content);
-
```
Comment (TomPretty) [inline support-frontend/assets/helpers/rendering/ssrPages.ts:6]: Didn't seem to be that useful a function!

File: support-frontend/assets/pages/paper-subscription-landing/paperSubscriptionLandingPage.tsx [hunk 3]
```diff
@@ -61,11 +67,7 @@ const PaperLandingPage = ({
 			product: 'Paper',
 			componentType: 'ACQUISITIONS_BUTTON',
 		})();
-		window.history.replaceState(
-			{},
-			null,
-			paperSubsUrl(newTab === HomeDelivery),
-		);
+		window.history.replaceState({}, '', paperSubsUrl(newTab === HomeDelivery));
```
Comment (TomPretty) [inline support-frontend/assets/pages/paper-subscription-landing/paperSubscriptionLandingPage.tsx:70]: the middle param is unused (it's literally called that in the type signature) but it should be passed an empty string https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 4]
```diff
@@ -50,13 +50,6 @@ export interface StripePaymentRequestButtonData {
 	paymentError: ErrorReason | null;
 }
 
-export interface Stripe3DSResult {
-	error?: Record<string, any>;
-	paymentIntent: {
-		id: string;
-	};
-}
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:58]: I was able to swap out all uses of this type with stripe's `PaymentIntentResult` type, which is just a better version of this one!

**Review by GHaberis — APPROVED** [review]
Looks good 👍

**Pull request merged**